### DISCUSSION TITLE
Alternatives to closing a PR when it's base branch is merged

### DISCUSSION THREAD

**Discussion by mkasberg** [community/community#6125] [Pull Requests]
I searched for any previous discussion of this, and I think the problem was summarized very well by @the9000 in this unofficial location: https://github.com/isaacs/github/issues/1439

To quote some of that here:
> ## Normal flow of events
>
> -   A trivial change, like a refactoring / renaming, is on the branch branch-renaming.
> -   A non-trivial change that takes a long time to develop and review, is on the branch branch-feature.
> -   To avoid back-fitting the renamed identifiers to the big feature change, branch-feature is not based on main, but based on branch-renaming, so it can continue while the renaming is being approved.
> -   There's a PR from branch-renaming to main.
> -   There's a PR from branch-feature to branch-renaming, with much discussion and updates doing on.
>
> ## The problem
>
> -   The PR for renaming is approved; branch-renaming is merged into main and deleted.
> -   The PR for the feature is now missing the base branch, so it gets closed, irrevocably.
>
> ## Proposed solution
>
> ```
> if 
>   a PR has a base branch `b` and 
>   branch `b` is merged into another branch `c`
> then 
>   set the PR's base branch to `c` // instead of closing it.
> ```
>
> (In most cases, c == main, but this case is not logically special.)

This description of the problem is spot-on, and highlights the difficulty of losing the information in the automatically-closed PR.

I think the proposed solution is a good one, but there are also other possible solutions -- such as simply _allowing_ the PR to be reopened with a different base branch.

**Comment by snowe2010** [comment]
How is this still an issue? This is absolutely the worst behavior imaginable.
